src/osd/modules/render/drawbgfx.cpp

```diff
@@ -423,10 +449,11 @@ bool video_bgfx::set_platform_data(bgfx::PlatformData &platform_data, osd_window
 		platform_data.nwh = wmi.info.cocoa.window;
 		break;
 #endif
-#if defined(SDL_VIDEO_DRIVER_WAYLAND) && SDL_VERSION_ATLEAST(2, 0, 16)
+#if defined(SDL_VIDEO_DRIVER_WAYLAND) && SDL_VERSION_ATLEAST(2, 0, 16) && defined(SDLMAME_USE_WAYLAND)
 	case SDL_SYSWM_WAYLAND:
 		platform_data.ndt = wmi.info.wl.display;
-		platform_data.nwh = wmi.info.wl.egl_window;
+		platform_data.nwh = (void *)create_wl_egl_window(dynamic_cast<sdl_window_info const &>(window).platform_window(), wmi.info.wl.surface);
+		platform_data.type = bgfx::NativeWindowHandleType::Wayland;
```

Comment:
Why are you explicitly casting to `void *`?  You don’t seem to be casting away CV qualifiers, and you aren’t doing a dynamic cast to the canonical object pointer.  You should be able to implicitly convert to `void *` if you just want the pointer value to be preserved.

Reply:
Implicit conversion seems to be working, will fix it.